lib/std/fs.zig

```diff
@@ -920,6 +920,13 @@ pub const Dir = struct {
         }
     }
 
+    // TODO: add support for e.g. AT_NO_AUTOMOUNT
+    const StatOptions = struct {};
+
+    pub fn statFile(self: Dir, sub_path: []const u8, statOptions: StatOptions) os.FStatAtError!os.Stat {
+        return os.fstatat(self.fd, sub_path, 0);
+    }
+
```

Comment:
Can you remove this part from the PR? `fs.File` is a cross platform abstraction and it already has a `stat` function. If you want to do the posix stat on a File, then the callsite should do that explicitly:

`std.os.fstatat(file.fd, sub_path, flags)`

Reply:
Its not uncommon to want to `stat` a file without opening it, e.g. imagine implementing `ls -l`. Note that you might not have permissions to open it!

Reply:
I've removed it from the PR for now.

Reply:
Yeah this function should exist but it should return fs.File.Stat, not os.Stat. Let's solve that separately